sys/cheri/cheric.h

```diff
@@ -243,26 +303,31 @@ cheri_bytes_remaining(const void * __capability cap)
 #define CHERI_FPRINT_PTR(f, ptr)					\
 	fprintf(f, _CHERI_PRINT_PTR_FMT(ptr))
 
-#endif	/* __has_feature(capabilities) */
-
-/* Allow use of some cheri_ptr macros in the purecap kernel
- * without extra ifdefs.
- * These become a no-op when compiling an hybrid kernel or a
- * normal kernel.
- */
-#ifdef CHERI_PURECAP_KERNEL
-
-#define cheri_bound(ptr, size) cheri_ptr((ptr), size)
-#define cheri_perm(ptr, size, perm) cheri_ptrperm((ptr), size, perm)
-#define cheri_valid(ptr) (cheri_gettag((ptr)) == 1)
-
-#else /* ! CHERI_PURECAP_KERNEL */
-
-#define cheri_bound(ptr, size) (ptr)
-#define cheri_perm(ptr, size, perm) (ptr)
-#define cheri_valid(ptr) (1)
-
-#endif /* ! CHERI_PURECAP_KERNEL */
+#else /* ! __has_feature(capabilities) */
+#ifdef _KERNEL
+#define	CHERI_ASSERT_VALID(ptr)
+#define	CHERI_ASSERT_BOUNDS(ptr, expect)
+#define	CHERI_ASSERT_EXBOUNDS(ptr, len)
+#define	CHERI_ASSERT_PTRSIZE_BOUNDS(ptr)
+#endif
+#endif /* ! __has_feature(capabilities) */
+
+#if defined(_KERNEL) && defined(CHERI_PURECAP_KERNEL)
```

Comment:
I would probably do `__CHERI_PURE_CAPABILITY__` here instead of `CHERI_PURECAP_KERNEL`.  Just as we have mostly replaced `CPU_CHERI` with `__has_feature(capabilities)`, we should really be using the compiler-provided macros to select ABIs when possible.

Reply:
I agree, I thought I already changed it but may have slipped through.